### PR TITLE
[10.x] Claco form refactoring

### PR DESCRIPTION
**Pull request by kitan1982**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no

### CODE REVIEW

File: main/core/Resources/modules/layout/modal/components/base.jsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ const BaseModal = props =>
             <span className={classes('modal-icon', props.icon)} />
           }
 
-          {props.title}
+          <div dangerouslySetInnerHTML={{__html: props.title}}></div>
```
Comment (Elorfin) [inline main/core/Resources/modules/layout/modal/components/base.jsx]: Don't put HTML in modal title. If you really need it, at least create a new modal.

Also note you should use the component `HtmlText` in `layout/components` to render HTML from string.

File: main/core/Resources/modules/layout/modal/components/user-picker.jsx [hunk 2]
```diff
@@ -9,7 +9,10 @@ import {UserTypeahead} from '#/main/core/layout/typeahead/users/typeahead.jsx'
 const UserPickerModal = props =>
   <BaseModal {...props}>
     <Modal.Body>
-      <UserTypeahead {...props} />
+      {props.content &&
+        <div dangerouslySetInnerHTML={{__html: props.content}}></div>
+      }
+      <UserTypeahead {...props}/>
```
Comment (Elorfin) [inline main/core/Resources/modules/layout/modal/components/user-picker.jsx]: Same here. This should not appear in base component.

File: plugin/claco-form/API/Finder/EntryFinder.php [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+    /**
+     * EntryFinder constructor.
+     *
+     * @DI\InjectParams({
+     *     "container" = @DI\Inject("service_container")
+     * })
+     *
+     * @param ContainerInterface $container
+     */
+    public function __construct(ContainerInterface $container)
```
Comment (Elorfin) [inline plugin/claco-form/API/Finder/EntryFinder.php]: You should only inject the services you need, not the whole container.

File: plugin/claco-form/API/Finder/EntryFinder.php [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+        $translator = $this->container->get('translator');
+        $currentUser = $tokenStorage->getToken()->getUser();
+
+        $isAnon = $currentUser === 'anon.';
+        $clacoForm = $clacoFormManager->getClacoFormById($extraData['clacoForm']);
+        $canEdit = $clacoFormManager->hasRight($clacoForm, 'EDIT');
+        $isCategoryManager = !$isAnon && $clacoFormManager->isCategoryManager($clacoForm, $currentUser);
+        $searchEnabled = $clacoForm->getSearchEnabled();
+        $sortBy = isset($extraData['sortBy']) ? $extraData['sortBy'] : null;
+        $direction = isset($extraData['direction']) ? $extraData['direction'] : null;
```
Comment (Elorfin) [inline plugin/claco-form/API/Finder/EntryFinder.php]: This is automatically done by the Finder. You don't have to handle it here. And so you can remove the new param you've added `extraData`.

```
$this->finder->search(
                'Claroline\CoreBundle\Entity\Theme\Theme', [
                    'limit' => 20,
                    'sortBy' => 'name', // <----------------- for DESC, use '-name'
                ]
            )
```

File: main/core/Resources/modules/layout/modal/components/base.jsx (hunk 1 above)
Comment (kitan1982) [reply main/core/Resources/modules/layout/modal/components/base.jsx]: So I will create a new modal

File: plugin/claco-form/API/Finder/EntryFinder.php (hunk 3 above)
Comment (kitan1982) [reply plugin/claco-form/API/Finder/EntryFinder.php]: The service I need generates a cyclic exception but I will try to do something

**Pull request merged**